From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 07 Mar 2018 17:24:00 +0200
Subject: Don't fetch grpc gem from RubyGems

Grpc gem is total shit, doesn't build from source without patching,
embeds three third-party libs including Google's very own fork of
OpenSSL and root CAs, contains a lot of bloat etc.

Therefore I install this shit from my Alpine package and
copy it into vendor/ before running bundler.

diff --git a/Gemfile b/Gemfile
--- a/ruby/Gemfile
+++ b/ruby/Gemfile
@@ -11,7 +11,6 @@
 gem 'rdoc', '~> 4.2'
 gem 'gitlab-gollum-lib', '~> 4.2', require: false
 gem 'gitlab-gollum-rugged_adapter', '~> 0.4.4', require: false
-gem 'grpc', '~> 1.15.0'
 gem 'sentry-raven', '~> 2.7.2', require: false
 gem 'faraday', '~> 0.12'
 
@@ -20,6 +19,9 @@
 gem 'licensee', '~> 8.9.0'
 
 gem 'google-protobuf', '~> 3.6'
+
+# This Google gem is totally messed, we handle it specially.
+gem 'grpc', path: 'vendor/grpc', require: false
 
 group :development, :test do
   gem 'rubocop', '~> 0.50', require: false
diff --git a/Gemfile.lock b/Gemfile.lock
--- a/ruby/Gemfile.lock
+++ b/ruby/Gemfile.lock
@@ -1,3 +1,9 @@
+PATH
+  remote: vendor/grpc
+  specs:
+    grpc (1.15.0)
+      google-protobuf (~> 3.1)
+
 GEM
   remote: https://rubygems.org/
   specs:
@@ -66,9 +72,6 @@
     google-protobuf (3.6.1)
     googleapis-common-protos-types (1.0.2)
       google-protobuf (~> 3.0)
-    grpc (1.15.0)
-      google-protobuf (~> 3.1)
-      googleapis-common-protos-types (~> 1.0.0)
     hashdiff (0.3.8)
     i18n (1.1.1)
       concurrent-ruby (~> 1.0)
@@ -184,7 +187,7 @@
   gitlab-gollum-rugged_adapter (~> 0.4.4)
   gitlab-markup (~> 1.6.5)
   google-protobuf (~> 3.6)
-  grpc (~> 1.15.0)
+  grpc!
   licensee (~> 8.9.0)
   listen (~> 0.5.0)
   rbtrace
